ada_feeding/ada_feeding/watchdog/ft_sensor_condition.py

```diff
@@ -101,7 +101,11 @@ def __ft_sensor_callback(self, msg: WrenchStamped) -> None:
                 msg.wrench.torque.z,
             ]
         )
-        ft_time = Time.from_msg(msg.header.stamp)
+        # We use the time at which the F/T message was received when checking
+        # how long has passed since receiving a message. This is to account
+        # for possible synchroneity issues or latency between the F/T node
+        # and the watchdog node.
+        ft_time = self._node.get_clock().now()
```

Comment:
I'm not 100% sure this is a good idea. Is this to capture a transport delay in the FT data and then handle the case where a whole bunch of messages are queued up at once?

In that case, wouldn't the better approach be to just modify the QoS of the forque_sensor_hardware to only have one message in the queue at once? Best Effort style?

Having a stale F/T message able to feeding the watchdog could allow for dangerous force conditions to persist.

Reply:
Good point, I reverted it.

The reason I had made the change is I noticed one time, when the F/T sensor had been on for a while, when the messages were being received continuously but with an older timestamp. But you're right that we should understand the underlying cause for it as opposed to putting this hack-y patch.